StyleCop.Analyzers/StyleCop.Analyzers.Test/Verifiers/CodeFixVerifier.cs

```diff
@@ -235,11 +235,14 @@ private static Task<Document> GetFixAllAnalyzerSolutionAsync(ImmutableArray<Diag
                     break;
                 }
 
-                if (--numberOfIterations < 0)
+                if (numberOfIterations <= 0)
                 {
-                    Assert.True(false, "The upper limit for the number of fix all iterations was exceeded");
+                    // If we reached 0 we either have not fixed everything we want to fix or the code fix is not supposed to fix all violations.
+                    break;
                 }
```

Comment:
:question: Why is this change needed? The `AreDiagnosticsDifferent` check above should be sufficient in detecting that a stable situation has been reached.